### PR TITLE
Add support for more fields to Edge gateway configuration, add multiple subnet support

### PR DESCRIPTION
**Pull request by Didainius**
This PR:
* This adds settings/proves the capability of edge gateway being able to attach multiple subnets and sub-allocation pools  (https://github.com/terraform-providers/terraform-provider-vcd/issues/323 )
* Adds additional missing settings to edge gateway structure
* Converts some bool fields to pointers to better distinguish between empty value and false.

Note. Test suite passed on 9.5 and 10.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,13 @@
 * Increment vCD API version used from 27.0 to 29.0
     * Remove fields `VdcEnabled`, `VAppParentHREF`, `VAppParentName`, `HighestSupportedVersion`, `VmToolsVersion`, `TaskHREF`, `TaskStatusName`, `TaskDetails`, `TaskStatus` from `QueryResultVMRecordType`
     * Add fields `ID, Type, ContainerName, ContainerID, OwnerName, Owner, NetworkHref, IpAddress, CatalogName, VmToolsStatus, GcStatus, AutoUndeployDate, AutoDeleteDate, AutoUndeployNotified, AutoDeleteNotified, Link, MetaData` to `QueryResultVMRecordType`, `DistributedInterface` to `NetworkConfiguration` and `RegenerateBiosUuid` to `VMGeneralParams`
-    * Change to pointers `DistributedRoutingEnabled` in `GatewayConfiguration` and `DistributedInterface` in `NetworkConfiguration`
+    * Change to pointers `DistributedRoutingEnabled` in `GatewayConfiguration` and
+    `DistributedInterface` in `NetworkConfiguration`
+* Add new fields to type `GatewayConfiguration`: `FipsModeEnabled` -
```
Comment (lvirbalas) [inline CHANGELOG.md]: Says `fields` in plural, but only a single field is referenced.

File: types/v56/types.go [hunk 2]
```diff
@@ -1567,15 +1567,34 @@ type EdgeGateway struct {
...
-	HaEnabled                       bool               `xml:"HaEnabled,omitempty"`                       // True if this gateway is highly available. (Requires two vShield edge VMs.)
-	AdvancedNetworkingEnabled       bool               `xml:"AdvancedNetworkingEnabled,omitempty"`       // True if the gateway uses advanced networking
-	DistributedRoutingEnabled       *bool              `xml:"DistributedRoutingEnabled,omitempty"`       // True if gateway is attached to a Distributed Logical Router
-	UseDefaultRouteForDNSRelay      bool               `xml:"UseDefaultRouteForDnsRelay,omitempty"`      // True if the default gateway on the external network selected for default route should be used as the DNS relay.
+	// BackwardCompatibilityMode. Default is false. If set to true, will allow users to write firewall
+	// rules in the old 1.5 format. The new format does not require to use direction in firewall
+	// rules. Also, for firewall rules to allow NAT traffic the filter is applied on the original IP
+	// addresses. Once set to true cannot be reverted back to false.
+	BackwardCompatibilityMode bool `xml:"BackwardCompatibilityMode,omitempty"`
```
Comment (lvirbalas) [inline types/v56/types.go]: Just a note, +1 for this way of adding full docs to fields in types.

File: CHANGELOG.md (hunk 1 above)
Comment (Didainius) [reply CHANGELOG.md]: Indeed. Fixed.

File: types/v56/types.go [hunk 3]
```diff
@@ -1567,15 +1567,34 @@ type EdgeGateway struct {
...
+	// GatewayBackingConfig defines configuration of the vShield edge VM for this gateway. One of:
+	// compact, full.
+	GatewayBackingConfig string `xml:"GatewayBackingConfig"`
+	// GatewayInterfaces holds configuration for
```
Comment (vbauzysvmware) [inline types/v56/types.go]: unfinished sentece

**Pull request merged**